src/nav/navReducer.js

```diff
@@ -29,7 +30,7 @@ export const initialState = getStateForRoute('loading');
 export default (state: NavigationState = initialState, action: Action): NavigationState => {
   switch (action.type) {
     case INITIAL_FETCH_COMPLETE:
-      return state.routes[0].routeName === 'main' ? state : getStateForRoute('main');
+      return getCurrentRouteName() === 'main' ? state : getStateForRoute('main');
```

Comment:
This is actually changing the behavior here, I think -- the `[0]` was intentional. Probably deserved a comment, or something in the code making it explicit.

The idea of this code:
```js
      return state.routes[0].routeName === 'main' ? state : getStateForRoute('main');
```
was: if we're *anywhere in the normal UI of the app* when an initial fetch completes, then remain there. Only if we're elsewhere -- which I think is just a way of saying, only if we're on the loading screen -- do we navigate, and in that case go to the main screen.

So in particular if you reopen the app after it was in the background for a while, and you were on some conversation, then it shows a loading animation there for a while, and when the fetch completes it shows you the conversation with its data. Similarly if you launch the app by opening a notification.

Reply:
I could go further and just make it check whether it's on the loading screen?

Reply:
That'd probably be good; we'd just want to check to be confident that there isn't another case where we'd want this behavior (and where the current code is getting it.) Probably best as a separate commit, either within this branch or afterward.